### ISSUE TITLE
Failed to load python model inside Self-built Triton Server in Centos 7

### ISSUE THREAD

**Issue by chandrameenamohan** [triton-inference-server/server#2960]
**Description**

I built the triton server from source code. I used centos 7.9 and x86_64 architecture.
I used this command to built: `./build.py --cmake-dir=/home/git/server/build --build-dir=/tmp/citritonbuild --no-container-build --endpoint=http --endpoint=grpc --repo-tag=common:r21.04 --repo-tag=core:r21.04 --repo-tag=backend:r21.04 --repo-tag=thirdparty:r21.04 --backend=python:r21.04 --enable-logging --enable-stats --enable-tracing`

TritonServer and python backend have build successfully. 

I have used python backend example model: https://github.com/triton-inference-server/python_backend/tree/main/examples/pytorch for model loading.

`
[root@3e63db3d18d2 home]# tritonserver --model-repository `pwd`/models --log-verbose 1
I0604 11:32:19.545486 113 backend_factory.h:44] Create TritonBackendFactory
I0604 11:32:19.556887 113 model_repository_manager.cc:770] AsyncLoad() 'pytorch'
I0604 11:32:19.557080 113 model_repository_manager.cc:1009] TriggerNextAction() 'pytorch' version 1: 1
I0604 11:32:19.557122 113 model_repository_manager.cc:1047] Load() 'pytorch' version 1
I0604 11:32:19.557134 113 model_repository_manager.cc:1066] loading: pytorch:1
I0604 11:32:19.658996 113 model_repository_manager.cc:1126] CreateInferenceBackend() 'pytorch' version 1
I0604 11:32:19.661077 113 shared_library.cc:44] OpenLibraryHandle: /opt/tritonserver/backends/python/libtriton_python.so
I0604 11:32:19.664484 113 python.cc:998] 'python' TRITONBACKEND API version: 1.0
I0604 11:32:19.664858 113 python.cc:1018] backend configuration:
{}
I0604 11:32:19.665045 113 python.cc:1085] shm-default-bytes=67108864,shm-growth-bytes=67108864,stub-timeout-seconds=4
I0604 11:32:19.665963 113 python.cc:1124] TRITONBACKEND_ModelInitialize: pytorch (version 1)
I0604 11:32:19.672198 113 model_config_utils.cc:1545] ModelConfig 64-bit fields:
I0604 11:32:19.672394 113 model_config_utils.cc:1547] 	ModelConfig::dynamic_batching::default_queue_policy::default_timeout_microseconds
I0604 11:32:19.672436 113 model_config_utils.cc:1547] 	ModelConfig::dynamic_batching::max_queue_delay_microseconds
I0604 11:32:19.672850 113 model_config_utils.cc:1547] 	ModelConfig::dynamic_batching::priority_queue_policy::value::default_timeout_microseconds
I0604 11:32:19.672937 113 model_config_utils.cc:1547] 	ModelConfig::ensemble_scheduling::step::model_version
I0604 11:32:19.672987 113 model_config_utils.cc:1547] 	ModelConfig::input::dims
I0604 11:32:19.673203 113 model_config_utils.cc:1547] 	ModelConfig::input::reshape::shape
I0604 11:32:19.673367 113 model_config_utils.cc:1547] 	ModelConfig::model_warmup::inputs::value::dims
I0604 11:32:19.673418 113 model_config_utils.cc:1547] 	ModelConfig::optimization::cuda::graph_spec::graph_lower_bound::input::value::dim
I0604 11:32:19.673803 113 model_config_utils.cc:1547] 	ModelConfig::optimization::cuda::graph_spec::input::value::dim
I0604 11:32:19.673917 113 model_config_utils.cc:1547] 	ModelConfig::output::dims
I0604 11:32:19.674334 113 model_config_utils.cc:1547] 	ModelConfig::output::reshape::shape
I0604 11:32:19.674466 113 model_config_utils.cc:1547] 	ModelConfig::sequence_batching::direct::max_queue_delay_microseconds
I0604 11:32:19.674839 113 model_config_utils.cc:1547] 	ModelConfig::sequence_batching::max_sequence_idle_microseconds
I0604 11:32:19.674906 113 model_config_utils.cc:1547] 	ModelConfig::sequence_batching::oldest::max_queue_delay_microseconds
I0604 11:32:19.674963 113 model_config_utils.cc:1547] 	ModelConfig::version_policy::specific::versions
I0604 11:32:19.676462 113 python.cc:1173] TRITONBACKEND_ModelInstanceInitialize: pytorch_0 (CPU device 0)
I0604 11:32:19.676735 113 backend_model_instance.cc:72] Creating instance pytorch_0 on CPU using artifact ''
I0604 11:32:19.676838 113 python.cc:1193] TRITONBACKEND_ModelInstanceInitialize: instance initialization successful pytorch_0 (device 0)
I0604 11:35:43.678076 113 python.cc:1232] TRITONBACKEND_ModelInstanceFinalize: delete instance state
I0604 11:35:43.679269 113 python.cc:1149] TRITONBACKEND_ModelFinalize: delete model state
I0604 11:35:43.679373 113 triton_backend_manager.cc:92] unloading backend 'python'
I0604 11:35:43.679557 113 python.cc:1104] TRITONBACKEND_Finalize: Start
I0604 11:35:43.679576 113 python.cc:1109] TRITONBACKEND_Finalize: End
E0604 11:35:43.680001 113 model_repository_manager.cc:1243] failed to load 'pytorch' version 1: Internal: Failed to initialize model instance pytorch_0
I0604 11:35:43.680056 113 model_repository_manager.cc:1009] TriggerNextAction() 'pytorch' version 1: 0
I0604 11:35:43.680084 113 model_repository_manager.cc:1024] no next action, trigger OnComplete()
I0604 11:35:43.680350 113 model_repository_manager.cc:615] VersionStates() 'pytorch'
I0604 11:35:43.680576 113 model_repository_manager.cc:615] VersionStates() 'pytorch'
I0604 11:35:43.680854 113 server.cc:504] 
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I0604 11:35:43.680904 113 server.cc:543] 
+---------+------+--------+
| Backend | Path | Config |
+---------+------+--------+
+---------+------+--------+

I0604 11:35:43.680921 113 model_repository_manager.cc:591] BackendStates()
I0604 11:35:43.681004 113 server.cc:586] 
+---------+---------+----------------------------------------------------------------------+
| Model   | Version | Status                                                               |
+---------+---------+----------------------------------------------------------------------+
| pytorch | 1       | UNAVAILABLE: Internal: Failed to initialize model instance pytorch_0 |
+---------+---------+----------------------------------------------------------------------+

I0604 11:35:43.681183 113 tritonserver.cc:1658] 
+----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
| Option                           | Value                                                                                                                                      |
+----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
| server_id                        | triton                                                                                                                                     |
| server_version                   | 2.9.0                                                                                                                                      |
| server_extensions                | classification sequence model_repository model_repository(unload_dependents) schedule_policy model_configuration system_shared_memory cuda |
|                                  | _shared_memory binary_tensor_data statistics                                                                                               |
| model_repository_path[0]         | /home/models                                                                                                                               |
| model_control_mode               | MODE_NONE                                                                                                                                  |
| strict_model_config              | 1                                                                                                                                          |
| pinned_memory_pool_byte_size     | 268435456                                                                                                                                  |
| min_supported_compute_capability | 0.0                                                                                                                                        |
| strict_readiness                 | 1                                                                                                                                          |
| exit_timeout                     | 30                                                                                                                                         |
+----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+

I0604 11:35:43.681243 113 server.cc:234] Waiting for in-flight requests to complete.
I0604 11:35:43.681287 113 model_repository_manager.cc:715] AsyncUnload() 'pytorch'
I0604 11:35:43.681307 113 model_repository_manager.cc:1009] TriggerNextAction() 'pytorch' version 1: 2
I0604 11:35:43.681361 113 model_repository_manager.cc:1092] Unload() 'pytorch' version 1
I0604 11:35:43.681410 113 model_repository_manager.cc:555] LiveBackendStates()
I0604 11:35:43.681428 113 server.cc:249] Timeout 30: Found 0 live models and 0 in-flight non-inference requests
error: creating server: Internal - failed to load all models
`

I don't know what exact reason of this failure as error is not descriptive.

**Triton Information**
What version of Triton are you using?
21.04

Are you using the Triton container or did you build it yourself?
I build it myself.

**To Reproduce**
Steps to reproduce the behavior.

You use this docker file:

FROM centos:centos7.9.2009
RUN yum -y update
RUN yum install -y epel-release 
RUN yum install -y re && yum install -y re2-devel
RUN yum install -y centos-release-scl-rh
RUN yum install -y devtoolset-8-gcc devtoolset-8-gcc-c++
RUN ln -s /opt/rh/devtoolset-8/root/usr/bin/g++ /usr/bin/g++
RUN ln -s /opt/rh/devtoolset-8/root/usr/bin/c++ /usr/bin/c++
RUN ln -s /opt/rh/devtoolset-8/root/usr/bin/gcc /usr/bin/gcc
RUN yum install -y git 
RUN yum install -y vim
RUN yum install -y autoconf
RUN yum install -y automake
RUN yum install -y boost-devel
RUN yum install -y make
RUN yum install -y openssl-devel
RUN yum install -y python3-devel
RUN yum install -y wget
RUN wget https://download-ib01.fedoraproject.org/pub/epel/7/x86_64/Packages/r/rapidjson-devel-1.1.0-2.el7.noarch.rpm &&  yum install -y rapidjson-devel-1.1.0-2.el7.noarch.rpm
RUN wget http://yum.oracle.com/repo/OracleLinux/OL8/developer/EPEL/x86_64/getPackage/libb64-devel-1.2-6.el8.x86_64.rpm & yum install -y libb64-devel-1.2-6.el8.x86_64.rpm
RUN mkdir -p /home/git
RUN cd /home/git && git clone https://github.com/triton-inference-server/server.git
RUN cd /home && wget https://github.com/Kitware/CMake/releases/download/v3.19.8/cmake-3.19.8.tar.gz
RUN cd /home && tar -xf cmake-3.19.8.tar.gz
RUN cd /home/cmake-3.19.8 && ./bootstrap && make && make install
RUN cd /home/git/server && git checkout -b r21.04 origin/r21.04
RUN cd /home/git/server && ./build.py --cmake-dir=/home/git/server/build --build-dir=/tmp/citritonbuild --no-container-build --endpoint=http --endpoint=grpc --repo-tag=common:r21.04 --repo-tag=core:r21.04 --repo-tag=backend:r21.04 --repo-tag=thirdparty:r21.04 --backend=python:r21.04 --enable-logging --enable-stats --enable-tracing
RUN python3 -m pip install --upgrade pip
RUN pip install nvidia-pyindex
RUN pip install tritonclient[all]

Build using this command:
`docker build -t centos-tfs:latest .`

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

I used this python model: https://github.com/triton-inference-server/python_backend/tree/main/examples/pytorch

**Expected behavior**

Server should load and run the model successfully

**Comment by Tabrizian** [comment]
@chandrameenamohan Are you able to run tritonserver with other backend. It looks like the error is generated from https://github.com/triton-inference-server/python_backend/blob/30e2d8e6d6983dcbe4f1d058931126b8b4f10df7/src/python.cc#L900. Can you share the directory content of `/opt/tritonserver/backends/python`? It could be that the python backend stub is not compiled properly.

**Comment by chandrameenamohan** [comment]
```
[root@3e63db3d18d2 python]# pwd
/opt/tritonserver/backends/python
[root@3e63db3d18d2 python]# ls -lh
total 624K
-rwxr-xr-x 1 root root 361K Jun  4 02:10 libtriton_python.so
-rwxr-xr-x 1 root root 243K Jun  4 02:10 triton_python_backend_stub
-rw-r--r-- 1 root root  14K Jun  4 02:10 triton_python_backend_utils.py
[root@3e63db3d18d2 python]# 
```
This is the directory content.

**Comment by chandrameenamohan** [comment]
I have not tried with any other backend. ONNX, PyTorch and TensorFlow asks for CUDA. I don't want to install CUDA for now, so I am not sure which other backend I can play with.

**Comment by chandrameenamohan** [comment]
I have rebuild it and somehow it worked but I am getting different error now. I will post a new question for it.
